.github/workflows/tests_linux.yml

```diff
@@ -227,8 +226,7 @@ jobs:
       - name: Install lightning.qubit device
         run: |
           cd main
-          python setup.py build_ext -i --define="ENABLE_BLAS=ON;ENABLE_KOKKOS=OFF;ENABLE_PYTHON=ON;CMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION)"
-          pip install -e .
+          CMAKE_ARGS="-DENABLE_BLAS=ON -DENABLE_KOKKOS=OFF -DCMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION)" pip install -e . -vv
```

Comment:
Just to clarify, if we don't set the CMAKE_ARGS, the default options chosen will still be the same as we have currently?

As in, a user can still compile everything as it was before by running `python -m pip install -e .` ?

Reply:
Good point clarifying. So a user will likely compile everything as before, however some default options changed. `setup.py` contains some platform dependent logic that was (at least partly) introduced to fix some CI issues. Some default options were forced which caused some issues in certain environments (e.g. Conda-Forge). For example, `Python_EXECUTABLE` and `PYTHON_EXECUTABLE` were always specified (but didn't need to apparently). In this PR, we try to keep `setup.py`  to a minimum, moving defaults either to CMake lists or specifying more environment dependent options in the Github actions files. This way, if say CF or spack have `CXX` defined in some env, we let CMake discover and deal with it. If we want to build with Brew's Clang on MacOS, we directly pass the option in `CMAKE_ARGS` so we don't have to check if `brew` or `clang++` exists in `setup.py` and specify some fallback.

Reply:
Thanks Vincent. I'd rather the pip install maintains the same defaults as before if not specified. Ideally we should be able to reproduce the released wheel build options without providing CMAKE_ARGS up front. If. that isn't the case, I think we should revert the changes to this.

Reply:
`CMAKE_ARGS` is not required here, the changes could be reverted to be passed by`--define` as well, it would just take two lines instead of one. 
As far as wheels are concerned, the only place defaults might have changed is in the MacOS wheel builders below. For x86 for example, they are 
```
CMAKE_ARGS="-DCMAKE_CXX_COMPILER=$(which clang++) -DCMAKE_LINKER=$(dirname $(which clang++))/lld -DOpenMP_ROOT=$(brew --prefix libomp)"
```
Since these options depend on the build environment and how we want to build the wheel, I thought they would be better put explicitly here instead of `setup.py` where they may override good default options in other contexts and on other systems (e.g. those without `brew` or `clang++`). But if you don't agree, I can put those back.

Reply:
I'm actually going to check whether these are required anyways.

Reply:
After discussion, I'll keep some default, I'll move others. Let's close it for now.